src/trufflesom/interpreter/nodes/UninitializedMessageSendNode.java

```diff
@@ -55,12 +56,26 @@ private PreevaluatedExpression specialize(final Object[] arguments) {
     return makeGenericSend();
   }
 
-  private GenericMessageSendNode makeGenericSend() {
-    AbstractDispatchNode dispatch = new UninitializedDispatchNode(selector);
-    GenericMessageSendNode send = new GenericMessageSendNode(selector, argumentNodes,
-        dispatch).initialize(sourceCoord);
+  private AbstractMessageSendNode makeGenericSend() {
+    int numArgs = argumentNodes.length;
+    AbstractMessageSendNode send;
+    if (numArgs == 1) {
```

Comment:
Maybe a switch statement instead

Reply:
I tried it, but don't like it any better.
I fear, this kind of code would need a different language to look good...